pkgs/by-name/ot/otb/package.nix

```diff
@@ -0,0 +1,109 @@
+{
+  cmake,
+  fetchFromGitHub,
+  makeWrapper,
+  lib,
+  stdenv,
+  swig,
+  which,
+  boost,
+  curl,
+  gdal,
+  itk_4_13,
+  libsvm,
+  libgeotiff,
+  muparser,
+  muparserx,
+  opencv,
+  perl,
+  python3,
+  shark,
+  tinyxml,
+}:
+  stdenv.mkDerivation (finalAttrs: {
+    pname = "otb";
+    version = "9.0.0";
+
+    src = fetchFromGitHub {
+      owner = "orfeotoolbox";
+      repo = "otb";
+      rev = "${finalAttrs.version}";
+      hash = "sha256-Ut2aimQL6Reg62iceoaM7/nRuEV8PBWtOK7KFHKp0ws=";
+    };
+
+    nativeBuildInputs = [
+      cmake
+      makeWrapper
+      python3.pkgs.wrapPython
+      swig
+      which
+    ];
+
+    buildInputs = [
+      boost
+      curl
+      gdal
+      itk_4_13
+      libsvm
+      libgeotiff
+      muparser
+      muparserx
+      opencv
+      perl
+      python3
+      python3.pkgs.numpy
```

Comment:
I'm not sure, but you might want to consult to use `with python3Packages; [ numpy ];`

Reply:
@jackyliu16 Thanks. 

I have added `extraPythonPackages` option and then similar to what you suggested.  Do you think is this fine ?

example:
```
    pythonInputs = with python3.pkgs; [
      numpy
    ] ++ (extraPythonPackages python3.pkgs);
```

Reply:
@daspk04 To be honest, I'm not an expert on Python packaging.
What I say is base on searching of other existing packages. 
I searched carefully just now and **found that most of them actually use operations similar to yours** directly,
but `pkgs/by-name/op/openscad-unstable/package.nix` or `tools/text/gpt2tc/default.nix` filter by `rg numpy $(rg -l mkDerivation)` provide a different soluition in basically the same situation as you, which use mkDerivation instead of buildPythonPackages or buildPythonApplication to packing python package and use python library.

Reply:
Thanks @jackyliu16 !

I basically followed similar [here](https://github.com/imincik/geospatial-nix/pull/501/files#diff-54bc19ec14bda80b39099b1f8f3b3ec76284cf719537c260db62432eabab4c61). So I assume then this is fine!